modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/CreditRatesProvider.java

```diff
@@ -0,0 +1,649 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.credit.cds;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Map.Entry;
+
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutablePreBuild;
+import org.joda.beans.ImmutableValidator;
+import org.joda.beans.PropertyDefinition;
+
+import com.google.common.collect.ImmutableMap;
+import com.opengamma.strata.basics.StandardId;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.collect.tuple.Pair;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivities;
+import com.opengamma.strata.market.sensitivity.PointSensitivities;
+import com.opengamma.strata.market.sensitivity.PointSensitivity;
+import com.opengamma.strata.pricer.ZeroRateSensitivity;
+import com.opengamma.strata.pricer.rate.RatesProvider;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import org.joda.beans.Bean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+/**
+ * The rates provider, used to calculate analytic measures.
+ * <p>
+ * The primary usage of this provider is to price credit default swaps on a legal entity.
+ * This includes credit curves, discounting curves and recovery rate curves.
+ */
+@BeanDefinition
+public final class CreditRatesProvider
+    implements ImmutableBean, Serializable {
+
+  /**
+   * The valuation date.
+   * <p>
+   * All curves and other data items in this provider are calibrated for this date.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final LocalDate valuationDate;
+  /**
+   * The credit curves.
+   * <p>
+   * The curve data, predicting the survival probability, associated with each legal entity and currency.
+   */
+  @PropertyDefinition(validate = "notEmpty", get = "private")
+  private final ImmutableMap<Pair<StandardId, Currency>, CreditDiscountFactors> creditCurves;
+  /**
+   * The discounting curves.
+   * <p>
+   * The curve data, predicting the discount factor, associated with each currency.
+   */
+  @PropertyDefinition(validate = "notEmpty", get = "private")
+  private final ImmutableMap<Currency, CreditDiscountFactors> discountCurves;
```

Comment:
Seems like a normal `DiscountFactors` would be better here, as the key does not use the legal entity.